bundles/org.eclipse.swt/Eclipse SWT Browser/win32/org/eclipse/swt/browser/Edge.java

```diff
@@ -641,6 +641,76 @@ void setupBrowser(int hr, long pv) {
 	webView.AddHostObjectToScript("swt\0".toCharArray(), hostObj);
 	hostDisp.Release();
 
+	class MouseNavigationFunction extends BrowserFunction {
+		public MouseNavigationFunction (Browser browser, String name) {
+			super (browser, name);
+		}
+		@Override
+		public Object function (Object[] arguments) {
+			System.out.println(Arrays.toString(arguments));
```

Comment:
I receive the mouse events just fine, e.g. in the "internal web browser".

But for some reason I do not receive any events in the element info popup or the variable hover.

Reply:
The mouse events only seem to be processed when then content is retrieved from an "ordinary" URL. Whenever you set contents via `setText` (such as in the Javadoc view, in the Java element info popup, in the help) it does not work. It does not depend on how the widget is embedded: in the help view, the first page is custom content (also with a custom context menu), which does not report mouse events, while when navigating to other entries in the help (like from the "see also" list), mouse events will be reported.

So it seems to be some problem of registering the browser function for "custom content" pages. Is it even possible to register browser functions to them?

Reply:
That's what I thought as well at first, but it works perfectly fine with other `setText()`-based browser usages, so does the following example:

```java
public class Test {

	public static void main(String[] args) {
		Display display = Display.getDefault();
		Shell shell = new Shell ();
		shell.setLayout(new FillLayout());

		new Browser(shell, SWT.EDGE).setText("hi");

		shell.open ();
		while (!shell.isDisposed()) {
			if (!display.readAndDispatch ()) display.sleep ();
		}
		display.dispose ();
	}
}
```

I don't understand what the element info / hover does differently.

Reply:
Oh, easy one... It's simply that those use `Browser.setJavascriptEnabled(false)`

Reply:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/iwebview2settings#get_isscriptenabled:

> _This only affects scripts in the document; scripts injected with ExecuteScript will run even if script is disabled. It is true by default._

So we need to register those listeners on the `window`, not the `document`.

Reply:
Hmm, no dice. Doesn't make a difference... I guess this script-based approach won't be possible universally :(